### PR TITLE
Add EventCorrelatorOptions for event rate limiting

### PR DESCRIPTION
**Pull request by mrogers950**
**What this PR does / why we need it**:
With the client side rate limiting from #47367 we also need to be able to create event sinks with non-default cache, aggregation, and spam settings.

**Which issue this PR fixes**:

**Special notes for your reviewer**:
depends on #47367

**Release note**:

```NONE
```
@derekwaynecarr

### CODE REVIEW

**Comment by enj** [comment]
@smarterclayton @liggitt @derekwaynecarr @mml can we consider this for 1.8 now that https://github.com/kubernetes/kubernetes/pull/47367 is in the queue?

**Comment by smarterclayton** [comment]
Please rebase and retest for 1.9

File: staging/src/k8s.io/client-go/tools/record/events_cache.go [hunk 1]
```diff
@@ -331,43 +416,73 @@ type EventCorrelateResult struct {
...
 //     times.
-func NewEventCorrelator(clock clock.Clock) *EventCorrelator {
-	cacheSize := maxLruCacheEntries
+//   * A source may burst 25 events about an object, but has a refill rate budget
+//     per object of 1 event every 5 minutes to control long-tail of spam.
+func NewDefaultEventCorrelatorOptions() *EventCorrelatorOptions {
+	return NewEventCorrelatorOptions(maxLruCacheEntries, defaultSpamBurst, defaultSpamQPS, defaultAggregateMaxEvents, defaultAggregateIntervalInSeconds)
+}
+
+func NewEventCorrelatorOptions(cacheSize, spamBurst int, spamQPS float32, aggregateMax, aggregateInterval int) *EventCorrelatorOptions {
```
Comment (mikedanese) [inline staging/src/k8s.io/client-go/tools/record/events_cache.go]: This function seems not useful and will have to change every time we add a filed to EventCorrelatorOptions

**Comment by mrogers950** [comment]
@smarterclayton rebased.

**Comment by simo5** [comment]
@smarterclayton can you review this and approve ?

**Comment by mrogers950** [comment]
@smarterclayton , @derekwaynecarr does this PR need anything else before approval?

**Pull request closed without merging**